From cafb16b905d188d2c8cf11d60713d47c5605574d Mon Sep 17 00:00:00 2001
From: Daniel Wagner <daniel.wagner@bmw-carit.de>
Date: Thu, 27 Feb 2014 11:13:59 +0100
Subject: [PATCH 5/7] session: Add connect/disconnect services

---
 include/session.h |  3 +++
 src/session.c     | 15 +++++++++++++++
 2 files changed, 18 insertions(+)

diff --git a/include/session.h b/include/session.h
index 9832f25..3f66664 100644
--- a/include/session.h
+++ b/include/session.h
@@ -100,6 +100,9 @@ int connman_session_parse_bearers(const char *token, GSList **list);
 
 const char *connman_session_get_owner(struct connman_session *session);
 
+int connman_session_connect(struct connman_service *service);
+int connman_session_disconnect(struct connman_service *service);
+
 #ifdef __cplusplus
 }
 #endif
diff --git a/src/session.c b/src/session.c
index 70fdfeb..7c18988 100644
--- a/src/session.c
+++ b/src/session.c
@@ -1431,6 +1431,21 @@ int __connman_session_destroy(DBusMessage *msg)
 	return 0;
 }
 
+int connman_session_connect(struct connman_service *service)
+{
+	DBG("service %p name %s", service, __connman_service_get_name(service));
+
+	return __connman_service_connect(service,
+				CONNMAN_SERVICE_CONNECT_REASON_SESSION);
+}
+
+int connman_session_disconnect(struct connman_service *service)
+{
+	DBG("service %p", service);
+
+	return __connman_service_disconnect(service);
+}
+
 static enum connman_session_state service_to_session_state(
 					enum connman_service_state state)
 {
-- 
1.8.5.3

