From 0fc8a9b241fdcb10431912e85d9feb04c1898125 Mon Sep 17 00:00:00 2001
From: Daniel Wagner <daniel.wagner@bmw-carit.de>
Date: Wed, 5 Mar 2014 10:58:21 +0100
Subject: [PATCH 6/7] service: Add autoconnect getter

---
 include/service.h | 1 +
 src/service.c     | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/include/service.h b/include/service.h
index d418dd8..1eaccd2 100644
--- a/include/service.h
+++ b/include/service.h
@@ -125,6 +125,7 @@ char **connman_service_get_proxy_excludes(struct connman_service *service);
 const char *connman_service_get_proxy_url(struct connman_service *service);
 const char *connman_service_get_proxy_autoconfig(struct connman_service *service);
 bool connman_service_get_favorite(struct connman_service *service);
+bool connman_service_get_autoconnect(struct connman_service *service);
 
 struct connman_service *connman_service_lookup_from_network(struct connman_network *network);
 
diff --git a/src/service.c b/src/service.c
index 288c68e..5a8c550 100644
--- a/src/service.c
+++ b/src/service.c
@@ -4920,6 +4920,11 @@ bool connman_service_get_favorite(struct connman_service *service)
 	return service->favorite;
 }
 
+bool connman_service_get_autoconnect(struct connman_service *service)
+{
+	return service->autoconnect;
+}
+
 int __connman_service_set_immutable(struct connman_service *service,
 						bool immutable)
 {
-- 
1.8.5.3

